### PR TITLE
Added missing React options for i18next

### PR DESCRIPTION
**Pull request by mauricedb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://react.i18next.com/components/i18next-instance
- [x] ~Increase the version number in the header if appropriate.~
- [x] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

### CODE REVIEW

File: types/i18next/index.d.ts [hunk 1]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
     interface ReactOptions {
         /**
          * set to true if you like to wait for loaded in every translated hoc
+         * @default false
          */
         wait?: boolean;
         /**
          * set it to fallback to let passed namespaces to translated hoc act as fallbacks
+         * @default default
```
Comment (mxl) [inline types/i18next/index.d.ts]: It's better to place default string value in quotes.

File: types/i18next/index.d.ts [hunk 2]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
...
          */
         nsMode?: string;
         /**
          * set it to the default parent element created by the Trans component.
          * @default div
          */
         defaultTransParent?: string;
+        /**
+         * set which events trigger a rerender, can be set to false or string of events
+         * @default languageChanged loaded
```
Comment (mxl) [inline types/i18next/index.d.ts]: Also here.

File: types/i18next/index.d.ts [hunk 3]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
...
+         */
+        bindI18n?: string | boolean;
+        /**
+         * set which events on store trigger a rerender, can be set to false or string of events
+         * @default added removed
```
Comment (mxl) [inline types/i18next/index.d.ts]: And here.

File: types/i18next/index.d.ts [hunk 4]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
...
         /**
          * set it to the default parent element created by the Trans component.
          * @default div
          */
         defaultTransParent?: string;
+        /**
+         * set which events trigger a rerender, can be set to false or string of events
+         * @default languageChanged loaded
+         */
+        bindI18n?: string | boolean;
```
Comment (mxl) [inline types/i18next/index.d.ts]: It can be false or string so the type is `false | string`.

File: types/i18next/index.d.ts [hunk 5]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
...
+        /**
+         * set which events on store trigger a rerender, can be set to false or string of events
+         * @default added removed
+         */
+        bindStore?: string | boolean;
```
Comment (mxl) [inline types/i18next/index.d.ts]: Also `false | string`.

File: types/i18next/index.d.ts [hunk 6]
```diff
@@ -97,17 +97,29 @@ declare namespace i18next {
...
         /**
          * set to true if you like to wait for loaded in every translated hoc
+         * @default false
          */
         wait?: boolean;
         /**
          * set it to fallback to let passed namespaces to translated hoc act as fallbacks
+         * @default default
          */
         nsMode?: string;
```
Comment (mxl) [inline types/i18next/index.d.ts]: According to documentation its type is `'default' | 'fallback'`.

**Review by mxl — CHANGES_REQUESTED** [review]

**Comment by mauricedb** [comment]
@mxl Just updated the PR with the changes you requested.

**Review by mxl — APPROVED** [review]

**Pull request merged**